service/matching/taskListManager.go

```diff
@@ -309,13 +309,7 @@ func (c *taskListManagerImpl) AddTask(ctx context.Context, params addTaskParams)
 		return c.taskWriter.appendTask(params.execution, params.taskInfo)
 	})
 
-	if err != nil {
-		c.logger.Error("Failed to add task",
-			tag.Error(err),
-			tag.WorkflowTaskListName(c.taskListID.name),
-			tag.WorkflowTaskListType(c.taskListID.taskType),
-		)
-	} else {
+	if err == nil && !syncMatch {
```

Comment:
@davidporter-id-au the `!syncMatch` is the change I mentioned offline. With this change `taskReader` is not going to be signaled for successful sync matches because nothing is written to `taskWriter` for those cases

Reply:
Yeah, I think this is ok. I'd normally suggest pulling this into a separate PR, but it's small enough for it to not be a problem I think.

Reply:
This seemed like a missed case that can be fixed inline here but if it required additional testing etc. I'd track it in separate PR.